somaticseq/SSeq_tsv2vcf.py

```diff
@@ -0,0 +1,462 @@
+
+    return inputParameters
+
+
+def tsv2vcf(tsv_fn, vcf_fn, tools, pass_score=0.5, lowqual_score=0.1, hom_threshold=0.85, het_threshold=0.01, single_mode=False, paired_mode=True, normal_sample_name='NORMAL', tumor_sample_name='TUMOR', print_reject=True, phred_scaled=True):
+
+    try:
+        with open(os.path.join(PRE_DIR, "VERSION")) as version_info_file:
+            version_line = version_info_file.readline()
+    except FileNotFoundError:
+        version_line = '##SomaticSeq\n'
+    
+    nan = float('nan')
+    
+    tools_code = {'CGA':           'M',
+                  'MuTect':        'M',
+                  'MuTect2':       'M',
+                  'VarScan2':      'V',
+                  'JointSNVMix2':  'J',
+                  'SomaticSniper': 'S',
+                  'VarDict':       'D',
+                  'MuSE':          'U',
+                  'LoFreq':        'L',
+                  'Scalpel':       'P',
+                  'Strelka':       'K',
+                  'TNscope':       'T'}
+    
+    
+    mvjsdu = ''
+    for tool_i in tools:
+        assert tool_i in tools_code.keys()
+        mvjsdu = mvjsdu + tools_code[tool_i]
+    
+    total_num_tools = len(mvjsdu)
+    tool_string = ', '.join( tools )
+    
+    def p2phred(p, max_phred=float('nan')):
+        '''Convert p-value to Phred-scale quality score.'''
+        
+        if p == 0:
+            Q = max_phred        
+        elif p > 0:    
+            Q = -10 * math.log10(p)
+            if Q > max_phred:
+                Q = max_phred
+        elif p == 1:
+            Q = 0
+        elif math.isnan(p) or p<0:
+            Q = nan
+                    
+        return Q
+        
+    
+    def dp4_to_gt(ref_for, ref_rev, alt_for, alt_rev, hom_threshold=hom_threshold, het_threshold=het_threshold):
+        try:
+            ref_for = int(ref_for)
+        except ValueError:
+            ref_for = 0
+            
+        try:
+            ref_rev = int(ref_rev)
+        except ValueError:
+            ref_rev = 0
+            
+        try:
+            alt_for = int(alt_for)
+        except ValueError:
+            alt_for = 0
+            
+        try:
+            alt_rev = int(alt_rev)
+        except ValueError:
+            alt_rev = 0
+            
+        var_counts = alt_for + alt_rev
+        ref_counts = ref_for + ref_rev
+        
+        if ref_counts == var_counts == 0:
+            gt = './.'
+            
+        elif var_counts/(var_counts+ref_counts) > hom_threshold:
+            gt = '1/1'
+            
+        elif var_counts/(var_counts+ref_counts) >= het_threshold:
+            gt = '0/1'
+            
+        else:
+            gt = '0/0'
+            
+        return gt
+    
+    
+    
+    with open(tsv_fn) as tsv, open(vcf_fn, 'w') as vcf:
+        
+        # First line is a header:
+        tsv_i = tsv.readline().rstrip()
+        
+        tsv_header = tsv_i.split('\t')
```

Comment:
Have a look at [csv.DictReader](https://docs.python.org/3/library/csv.html#csv.DictReader), it will parse the row items into a dictionary based on the names in the header row.

Reply:
Good idea, but I think I'll put that off until the next iteration. Fear of introducing new bugs.